src/Psalm/Internal/Codebase/Populator.php

```diff
@@ -281,8 +316,65 @@ private function populateClassLikeStorage(ClassLikeStorage $storage, array $depe
     }
 
     private function populateOverriddenMethods(
-        ClassLikeStorage $storage
+        ClassLikeStorage $storage,
+        ClassLikeStorageProvider $storage_provider
     ): void {
```

Comment:
This was pulled out of `populateDataFromImplementedInterfaces` when I removed the loop from it, so that it could be done afterward. I think it's actually easier to understand this way, one issue that took me a while to figure out is that `$storage->class_implements` may already have had items set elsewhere (from Reflection I believe), so this is actually populating overridden methods that were set elsewhere as well as from this class.